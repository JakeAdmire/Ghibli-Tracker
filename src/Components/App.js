import React, { Component } from 'react';
import { Header } from './Header/Header';
import { Account } from './Account/Account';
import { CardContainer } from './CardContainer/CardContainer';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <Header />
          <Account />
          <CardContainer />
        </header>
      </div>
    );
  }
}

export default App;
