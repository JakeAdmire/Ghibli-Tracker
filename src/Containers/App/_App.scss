*:focus { outline: none; }

.App {
  display: flex;
  justify-content: center; 
  height: 100vh; 
  align-items: center; 
}

.login {
  position: absolute;
  top: 5px; right: 15px;
  transition: $ease;
  text-shadow: 0px 2px 3px #00000066;
  font-family: 'Poiret One';
  @include customFont(24px, initial, none, $vibrant-color);
  &:hover {
    text-decoration: underline;
    color: $vibrant-hover;
  }
  &:active {
    opacity: 0.5;
  }
}

.show-faves {
  border: none;
  background: none;

  position: absolute;
  top: 35px; right: 15px;

  transition: $ease;

  text-shadow: 0px 2px 3px #00000066;
  font-family: 'Poiret One';

  @include customFont(24px, initial, initial, $vibrant-color);

  &:hover {
    text-decoration: underline;
    cursor: pointer;
    color: $vibrant-hover;
  }

  &:active {
    opacity: 0.5;
  }

}