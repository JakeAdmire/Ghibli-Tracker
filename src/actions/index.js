export const toggleFavorited = (id) => ({
  type: 'TOGGLE_FAVORITED',
  id
})

export const setFilter = (filter) => ({
  type: 'SET_FILTER',
  filter
})

export const toggleInfo = (id) => ({
  type: 'TOGGLE_INFO',
  id
})

export const addFilms = (films) => ({
  type: 'ADD_FILMS',
  films
})

export const addUser = (name, email, password) => ({
  type: 'ADD_USER',
  name,
  email,
  password
})

export const loginUser = (email, password) => ({
  type: 'LOGIN_USER',
  email,
  password
})