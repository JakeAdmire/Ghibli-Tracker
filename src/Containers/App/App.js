import React, { Component } from 'react';
import { connect } from 'react-redux';
import { key } from '../../helpers/apiKey';
import { NavLink, Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import * as help from '../../helpers/fetch';
import Account from './Account/Account';
import CardContainer from '../../Components/CardContainer/CardContainer';
import { Header } from '../../Components/Header/Header';
import { Info } from '../../Components/CardContainer/Card/Info/Info';
import { addFilms } from '../../actions';


export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect: false
    }
  }

  componentDidMount() {
    this.buildCards()
  }

  buildCards = async() => {
    let url = `https://api.themoviedb.org/3/discover/movie?api_key=${key}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_companies=10342`;
    const recentFilms = await help.fetchFilms(url);
    this.props.addFilms(recentFilms.results)
  }

  determineFavorites = async () => {
    const { user } = this.props
    if (user.name) {
      const results = await help.fetchFilms(`http://localhost:3000/api/users/${user.id}/favorites`);
      if (JSON.stringify(this.props.films) === JSON.stringify(results.data)) {
        this.buildCards();
      } else {
        this.props.addFilms(results.data);
      }
    } else {
      this.setState({redirect: true})
    }
  }

  render() {
    return (
      <div className="App">
        <Header />
        <NavLink to='/login' className="login">Log in:</NavLink>
        <button onClick={this.determineFavorites}>FAVES</button>
        <Route exact path='/' component={CardContainer} />
        <Route exact path='/login' component={Account} />
        <Route path='/:id' render={ ({match}) => {
          const film = this.props.films.find(film => film.id == match.params.id);
          if (film) { return <Info {...film} /> }
        } }/>
        <Route exact path='/' render={() => (
          this.state.redirect && <Redirect to='/login' />
        )} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  films: state.films,
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  addFilms: (films) => dispatch(addFilms(films))
})

export default connect(mapStateToProps, mapDispatchToProps)(App)

App.propTypes = {
  addFilms: PropTypes.func.isRequired,
  films: PropTypes.array,
  user: PropTypes.object.isRequired
}