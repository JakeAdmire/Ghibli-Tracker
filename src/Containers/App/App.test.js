import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { addFilms } from '../../actions';


describe('App', () => {

  let wrapper;
  let mockRecentFilms;
  let mockFilms;
  let mockState;
  

  describe('App', () => {

    it('should call addFilms with correct arguments', () => {

      const props = {
        addFilms: jest.fn()
      }
  
      wrapper = shallow(<App {...props} />)
  
      mockRecentFilms = {
        "page": 1,
        "total_results": 60,
        "total_pages": 3,
        "results": [
            {
                "vote_count": 7068,
                "id": 129,
                "video": false,
                "vote_average": 8.5,
                "title": "Spirited Away",
                "popularity": 26.943,
                "poster_path": "/oRvMaJOmapypFUcQqpgHMZA6qL9.jpg",
                "original_language": "ja",
                "original_title": "千と千尋の神隠し",
                "genre_ids": [
                    16, 10751, 14
                ],
                "backdrop_path": "/mnpRKVSXBX6jb56nabvmGKA0Wig.jpg",
                "adult": false,
                "overview": "A young girl, Chihiro, becomes trapped in a strange new world of spirits. When her parents undergo a mysterious transformation, she must call upon the courage she never knew she had to free her family.",
                "release_date": "2001-07-20"
            }]
        }
  
      wrapper.instance().props.addFilms(mockRecentFilms.results)
      expect(props.addFilms).toHaveBeenCalledWith(mockRecentFilms.results)
    })
  
    it('should match the snapshot', () => {
      const props = { addFilms: jest.fn() }
      wrapper = shallow(<App {...props} />)
      expect(wrapper.debug()).toMatchSnapshot();
    })

  })

  describe('mapStateToProps', () => {

    it('should return an array of film objects as props', () => {

      mockFilms = [{
        "vote_count": 7068,
        "id": 129,
        "video": false,
        "vote_average": 8.5,
        "title": "Spirited Away",
        "popularity": 26.943,
        "poster_path": "/oRvMaJOmapypFUcQqpgHMZA6qL9.jpg",
        "original_language": "ja",
        "original_title": "千と千尋の神隠し",
        "genre_ids": [
            16, 10751, 14
        ],
        "backdrop_path": "/mnpRKVSXBX6jb56nabvmGKA0Wig.jpg",
        "adult": false,
        "overview": "A young girl, Chihiro, becomes trapped in a strange new world of spirits. When her parents undergo a mysterious transformation, she must call upon the courage she never knew she had to free her family.",
        "release_date": "2001-07-20"
      }]

      mockState = {
        films: mockFilms,
        filter: 'Show all',
        user: {}
      }

      const expected = {
        films: mockFilms,
        user: {}
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)

    })

  })

  describe('mapDispatchToProps', () => {

    it('should provide a method to use to dispatch an action creator', () => {
      const mockFilm = {title: 'Spirited Away', id: 129}
      const mockDispatch = jest.fn()
      const actionToDispatch = addFilms(mockFilm)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addFilms(mockFilm)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })

})
