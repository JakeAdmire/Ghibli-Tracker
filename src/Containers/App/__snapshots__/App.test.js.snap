// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match the snapshot 1`] = `
"<div className=\\"App\\">
  <Header />
  <NavLink to=\\"/login\\" className=\\"login\\">
    LOG IN:
  </NavLink>
  <button className=\\"show-faves\\" onClick={[Function]}>
    FAVES
  </button>
  <Route exact={true} path=\\"/\\" component={[Function: Connect]} />
  <Route exact={true} path=\\"/login\\" component={[Function: Connect]} />
  <Route path=\\"/:id\\" render={[Function: render]} />
  <Route exact={true} path=\\"/\\" render={[Function: render]} />
</div>"
`;
